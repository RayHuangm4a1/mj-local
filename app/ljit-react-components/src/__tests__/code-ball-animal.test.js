import React from 'react';
import { shallow, mount, } from 'enzyme';
import Animal from '../components/code-ball/animal';

const { SizeEnums, } = Animal;

describe('CodeBall Animal', () => {
	it('should handle default props', () => {
		const { size, } = Animal.defaultProps;

		expect(size).toEqual(SizeEnums.MEDIUM_25);
	});

	it('should render correctly', () => {
		const wrapper = shallow(<Animal text='0'/>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class "ljit-code-ball--animal"', () => {
		const wrapper = shallow(<Animal text='0'/>);

		expect(wrapper.hasClass('ljit-code-ball--animal')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<Animal text='0' className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const size = SizeEnums.MEDIUM_25;
		const text = '0';
		const className = 'mock-class';
		const style = { fontSize: '14px', };
		const wrapper = mount(
			<Animal
				size={size}
				text={text}
				className={className}
				style={style}
			/>
		);

		expect(wrapper.props().size).toEqual(size);
		expect(wrapper.props().text).toEqual(text);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().style).toMatchObject(style);
	});
});

describe('Size Enums', () => {
	it('should contain DEFAULT_12 property', () => {
		const typeName = 12;
		const formatType = 'DEFAULT_12';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contain SMALL_14 property', () => {
		const typeName = 14;
		const formatType = 'SMALL_14';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contain SMALL_20 property', () => {
		const typeName = 20;
		const formatType = 'SMALL_20';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contain MEDIUM_22 property', () => {
		const typeName = 22;
		const formatType = 'MEDIUM_22';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contain MEDIUM_24 property', () => {
		const typeName = 24;
		const formatType = 'MEDIUM_24';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contain MEDIUM_25 property', () => {
		const typeName = 25;
		const formatType = 'MEDIUM_25';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contain BIG_32 property', () => {
		const typeName = 32;
		const formatType = 'BIG_32';

		expect(SizeEnums[formatType]).toEqual(typeName);
	});
});
