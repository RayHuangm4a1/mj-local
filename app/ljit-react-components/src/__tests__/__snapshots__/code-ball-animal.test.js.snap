// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBall Animal should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Animal
    size={25}
    text="0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
          Object {
            "fontSize": "20px",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        üê≠
      </div>,
      "className": "ljit-code-ball ljit-code-ball--animal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "üê≠",
        "style": Object {
          "fontSize": "20px",
          "height": "25px",
          "width": "25px",
        },
      },
      "ref": null,
      "rendered": "üê≠",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
            Object {
              "fontSize": "20px",
              "height": "25px",
              "width": "25px",
            }
          }
        >
          üê≠
        </div>,
        "className": "ljit-code-ball ljit-code-ball--animal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "üê≠",
          "style": Object {
            "fontSize": "20px",
            "height": "25px",
            "width": "25px",
          },
        },
        "ref": null,
        "rendered": "üê≠",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
